webgpu.h

```diff
@@ -0,0 +1,1025 @@
+#ifndef WGPU_WGPU_H_
+#define WGPU_WGPU_H_
+
+#if !defined(WGPU_EXPORT)
+#    if defined(_WIN32)
+#        if defined(WGPU_IMPLEMENTATION)
+#            define WGPU_EXPORT __declspec(dllexport)
+#        else
+#            define WGPU_EXPORT __declspec(dllimport)
+#        endif
+#    else  // defined(_WIN32)
+#        if defined(WGPU_IMPLEMENTATION)
+#            define WGPU_EXPORT __attribute__((visibility("default")))
+#        else
+#            define WGPU_EXPORT
+#        endif
+#    endif  // defined(_WIN32)
+#endif  // defined(WGPU_EXPORT)
+
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+
+typedef struct wgpuBindGroupImpl* wgpuBindGroup;
+typedef struct wgpuBindGroupLayoutImpl* wgpuBindGroupLayout;
+typedef struct wgpuBufferImpl* wgpuBuffer;
+typedef struct wgpuCommandBufferImpl* wgpuCommandBuffer;
+typedef struct wgpuCommandEncoderImpl* wgpuCommandEncoder;
+typedef struct wgpuComputePassEncoderImpl* wgpuComputePassEncoder;
+typedef struct wgpuComputePipelineImpl* wgpuComputePipeline;
+typedef struct wgpuDeviceImpl* wgpuDevice;
+typedef struct wgpuFenceImpl* wgpuFence;
+typedef struct wgpuPipelineLayoutImpl* wgpuPipelineLayout;
+typedef struct wgpuQueueImpl* wgpuQueue;
+typedef struct wgpuRenderPassEncoderImpl* wgpuRenderPassEncoder;
+typedef struct wgpuRenderPipelineImpl* wgpuRenderPipeline;
+typedef struct wgpuSamplerImpl* wgpuSampler;
+typedef struct wgpuShaderModuleImpl* wgpuShaderModule;
+typedef struct wgpuTextureImpl* wgpuTexture;
+typedef struct wgpuTextureViewImpl* wgpuTextureView;
+
+typedef enum {
+    WGPU_ADDRESS_MODE_REPEAT = 0x00000000,
+    WGPU_ADDRESS_MODE_MIRROR_REPEAT = 0x00000001,
+    WGPU_ADDRESS_MODE_CLAMP_TO_EDGE = 0x00000002,
+    WGPU_ADDRESS_MODE_FORCE32 = 0x7FFFFFFF
+} wgpuAddressMode;
```

Comment:
Sure, what naming conventions would you like for enums, bitfields, structs, callbacks and API function pointers? We can discuss it here and it will be easy to change the code generator for this initial version.

Reply:
Let's start the type names with WGPU, i.e. `WGPUAddressMode`. I know it's not pretty, but the alternatives seem to be worse.

Reply:
I'll give it a try.

Reply:
Updating the casing, PTAL. I think it looks ok.